src/utils.coffee

```diff
@@ -18,4 +18,17 @@ class Utils
     return Math.random() unless seed
     # a MUCH simplified version inspired by PlanOut.js
     parseInt(@sha1(seed).substr(0, 13), 16) / 0xFFFFFFFFFFFFF
+  @checkWeights: (variants) ->
+    contains_weight = []
+    contains_weight.push(value.weight?) for key, value of variants
+    contains_weight.every (has_weight) -> has_weight
+  @sumWeights: (variants) ->
```

Comment:
I would probably express this as

```coffee
variants.reduce ((total, variant) -> total + (variant.weight || 0)), 0
```

it's slightly more dense, but it's "just" a sum of weights...